src/main/java/org/jacp/repositry/CompetitionRepository.java

```diff
@@ -2,13 +2,20 @@
 
 import org.jacp.entity.CompetitionEntity;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.util.List;
+import java.util.Optional;
+
 /**
  * @author saffchen created on 04.12.2023
  */
 @Transactional
 @Repository
 public interface CompetitionRepository extends JpaRepository<CompetitionEntity, Long> {
+
+    @Query("select ce from CompetitionEntity ce where ce.status = 'QUEUED'")
+    Optional<List<CompetitionEntity>> getAllQueued();
```

Comment:
Minor: it's better to add flexibility to the method: getAllByStatus(String status) , then you will be able to filter by not only the QUEUED status.

Reply:
Fixed